vector-im/element-x-android
# [Message Actions] Retry sending failed messages

## Changes

- Added a retry sending menu that appears when tapping on the timestamp view of a failed to send message: it allows you to retry sending the message or delete them from your local echoes.
- Fixed an issue that retrieved the original state of the event instead of the current one when using this feature, so the `sendState` sometimes wasn't updated.
- Added extra touch area to the timestamp.

## Why

Fixes #487.

## Test

- Open a room.
- Enable airplane mode (disable any network connection).
- Send a couple of messages.
- Disable airplane mode, wait until you have internet connection.
- Retry sending one of the messages and removing the other one.

It could be interesting to also try re-sending the message with no network connection, although it worked fine in my tests.

**Note:** there is an issue currently that causes the sliding sync loop to be stopped when network connection is lost and it never recovers. This means that sending messages from the previously offline device works, but you won't see any incoming messages or the room list updated.
